include/engine/guidance/route_step.hpp

```diff
@@ -172,16 +179,32 @@ inline RouteStep &RouteStep::AdaptStepSignage(const RouteStep &origin)
     return *this;
 }
 
+inline bool RouteStep::hasLanesAtTurn() const { return NumLanesInTotal() != 0; }
+
+inline LaneID RouteStep::NumLanesInTurn() const
+{
+    return intersections.front().lanes.lanes_in_turn;
+}
+
+inline LaneID RouteStep::NumLanesInTotal() const
+{
+    return intersections.front().lane_description.size();
+}
+
 inline LaneID RouteStep::NumLanesToTheRight() const
 {
+    if (!hasLanesAtTurn())
+        return 0;
+
     return intersections.front().lanes.first_lane_from_the_right;
 }
 
 inline LaneID RouteStep::NumLanesToTheLeft() const
 {
-    LaneID const total = intersections.front().lane_description.size();
-    return total - (intersections.front().lanes.lanes_in_turn +
-                    intersections.front().lanes.first_lane_from_the_right);
+    if (!hasLanesAtTurn())
+        return 0;
+
+    return NumLanesInTotal() - (NumLanesInTurn() + NumLanesToTheRight());
```

Comment:
Hm it's a uint8_t so it will never be negative - I could add an assert for lhs > rhs - not sure if this could ever happen (.lane_description out of sync with .lanes), so good to assert from my side. cc @MoKob